### DISCUSSION TITLE
difficulties building under MacOS Ventura

### DISCUSSION THREAD

**Discussion by steve-summit** [openssl/openssl#22073] [Q&A]
Trying to build under MacOS 13.0 Ventura, using clang version 14.0.3.

Configure ran fine.  But the build generates thousands of errors, chief among them:
* include/openssl/macros.h:155: error: "OPENSSL_API_COMPAT expresses an impossible API compatibility level"
* include/internal/cryptlib.h:62: warning: type specifier missing, defaults to 'int' [for the line `DEFINE_LHASH_OF_EX(MEM);`, suggesting that the `DEFINE_LHASH_OF_EX` is not expanding correctly]
* error: unknown type name 'OSSL_LIB_CTX'

Happy to run experiments, provide more info, etc.

Config info:
[config_dump.txt](https://github.com/openssl/openssl/files/12581242/config_dump.txt)

**Comment by steve-summit** [comment]
Whoops... looks like it was an obscure issue in my compilation environment.  Closing.

**Comment by tom-cosgrove-arm** [comment]
Glad you've got it sorted.

If you're able to give a quick summary of the problem in your environment, and what you did to fix it, this could help people in the future who have the same problem

**Comment by steve-summit** [comment]
@Tom-cosgrove-arm: I'm going to investigate further, but it's unlikely to be useful to anyone else!

The problem was that in my environment, `cc` is my own, custom wrapper around clang, allowing me to customize a few options to my taste.  I _thought_ the only options I was customizing had to do with warning levels and output colorization, so they weren't supposed to have any impact on the language or the code accepted.  They certainly weren't supposed to turn a well-written, portable codebase like openssl into a roaring quagmire of 10,000 error messages!  Yet somehow they did.  Very strange.

(If anything I wasted the openssl GitHub page's time with my purely idiosyncratic problem.  But this thread did end up serving as a highly effective [rubber duck](https://en.wikipedia.org/wiki/Rubber_duck_debugging): Less than 10 minutes after posting here, I'd figured out and solved my problem! :-) )
